### ISSUE TITLE
iconv: convert to HZ encoding won't escape '~'.

### ISSUE THREAD

**Issue by ikazont** [ikazont/netbsd#139]
**[Original report](https://bitbucket.org/tnozaki/netbsd/issues/139) by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

* Citrus iconv

```
printf '~' | iconv -f UTF-8 -t hz
~
```

* Perl piconv

```
printf '~' | piconv -f UTF-8 -t hz
~~
```

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

hmm, GNU libiconv has same problem too?

or this is correct behavior? i can’t remember anymore…

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

CLEANUP/ENHANCEMENT/BUGFIX: rewrite citrus_hz module:
  1. slight performance improvement, drop incomplete hz+/ehz encoding support.
  2. split citrus_hz's mbrtowc/wcrtomb -> decoder/encoder.
  3. fix Issue #139 - iconv: convert to HZ encoding won't escape '~'.
